File: fsharp/lasagna.fs
module LuciansLusciousLasagna

let layerPreparationTime = 2

let expectedMinutesInOven = 40

let remainingMinutesInOven inOven = expectedMinutesInOven - inOven

let preparationTimeInMinutes layers = layers * layerPreparationTime

let elapsedTimeInMinutes layers timeInOven = preparationTimeInMinutes layers + timeInOven

File: csharp/lasagna.cs
class Lasagna
{
    
    public int ExpectedMinutesInOven() => 40;
    
    public int RemainingMinutesInOven(int spentInOven)
        => ExpectedMinutesInOven() - spentInOven;
    
    public int PreparationTimeInMinutes(int layers)
        => 2 * layers;
    
    public int ElapsedTimeInMinutes(int layers, int spentInOven)
        => PreparationTimeInMinutes(layers) + spentInOven;
}